jerry-core/ecma/base/ecma-globals.h

```diff
@@ -690,15 +690,31 @@ typedef enum
 } ecma_string_container_t;
 
 /**
+ * ECMA string related constants
+ */
+typedef enum
+{
+  ECMA_STRING_CONTAINER_MASK = 0x7u, /**< container mask */
+  ECMA_STRING_REF_ONE = 1u << 3, /**< value for increasing or decreasing the reference counter */
+  ECMA_STRING_MAX_REF = 0x1fffu << 3, /**< maximum value of the reference counter (8191) */
+} ecma_string_constants_t;
```

Comment:
I did search for the enums. There are exactly 49 enum typedefs in jerry-core. Wherever they occur, they are used for grouping a set of values we're choosing one from, or a set of flags we are bit-or-ing together. They are almost always also used as types in data structures, local variables, function arguments, or return values. In some rare cases only their values (labels) are used, because we chose not to use their real type in a struct but went for a type with an exactly known size (like `uint8_t`). (Note: this is going against the type safety you also mentioned above.) I haven't seen any use where a bit mask, a starting counter, and a counter limit would be grouped together in an enum.

(Case-by-case analysis: `jerry_flag_t`, `jerry_fatal_code_t` are used as types in the code. `ecma_property_flags_t` is not used as a type, since the struct that countains it forces the flags into a `unit8_t`. `vm_oc_get_types` is used in a similar way, it is used as a set of values, only the real type used in the code is different by some reason. `re_flags_t` is not used as a type either, but it could be -- there are already some code paths in js-lexer.c which cast around 16 and 32 bit integers when dealing with these flags because of no apparent reason.)